packages/sit-onyx/src/components/OnyxDialog/OnyxDialog.vue

```diff
@@ -0,0 +1,125 @@
+<script lang="ts" setup>
+import { ref, watch } from "vue";
+import { useDensity } from "../../composables/density";
+import type { OnyxDialogProps } from "./types";
+
+const props = withDefaults(defineProps<OnyxDialogProps>(), {
+  open: false,
+  modal: false,
+  alert: false,
+});
+
+const emit = defineEmits<{
+  /**
+   * Emitted when the dialog should be closed (only when `modal` property is `true`).
+   */
+  close: [];
+}>();
+
+defineSlots<{
+  /**
+   * Dialog content. For accessibility purposes it is strongly recommended
+   * to focus an element inside the dialog when it opens. The element will depend
+   * on your use case (e.g. an input, button etc.).
+   */
+  default(): unknown;
+}>();
+
+const dialogRef = ref<HTMLDialogElement>();
+const { densityClass } = useDensity(props);
+
+/**
+ * Shows the dialog either as default dialog or modal.
+ */
+const openDialog = () => {
+  if (props.modal) dialogRef.value?.showModal();
+  else dialogRef.value?.show();
+};
+
+// sync open state
+watch(
+  [dialogRef, () => props.open],
+  () => {
+    if (props.open) openDialog();
+    else dialogRef.value?.close();
+  },
+  { immediate: true },
+);
```

Comment:
`immediate` doesn't have an effect here, as dialogRef will always be undefined initially.

```suggestion
watch(
  [dialogRef, () => props.open],
  () => {
    if (props.open) openDialog();
    else dialogRef.value?.close();
  }
);
```

Reply:
I am not sure if this is also true for server side rendering / static site generation in Nuxt

Reply:
Template refs will only be available after the component is mounted. So on the server it will also be undefined